core/chains/evm/client/errors.go

```diff
@@ -233,17 +235,62 @@ var zkEvm = ClientErrors{
 	OutOfCounters: regexp.MustCompile(`(?:: |^)not enough .* counters to continue the execution$`),
 }
 
-var clients = []ClientErrors{parity, geth, arbitrum, metis, substrate, avalanche, nethermind, harmony, besu, erigon, klaytn, celo, zkSync, zkEvm}
+var clientsLock = sync.Mutex{}
+var clients = map[string]ClientErrors{
+	"parity":     parity,
+	"geth":       geth,
+	"arbitrum":   arbitrum,
+	"metis":      metis,
+	"substrate":  substrate,
+	"avalanche":  avalanche,
+	"nethermind": nethermind,
+	"harmony":    harmony,
+	"besu":       besu,
+	"erigon":     erigon,
+	"klaytn":     klaytn,
+	"celo":       celo,
+	"zkSync":     zkSync,
+	"zkEvm":      zkEvm,
+}
+
+// SetClientErrorRegexes is called on startup to set the client errors from the config
+func SetClientErrorRegexes(errsRegex config.ClientErrors) {
+	clientsLock.Lock()
+	defer clientsLock.Unlock()
+	clients["tomlConfig"] = ClientErrors{
```

Comment:
We cannot have global state. A node connects to multiple chains, so they would overwrite one another. It undermines testability besides - even if this worked with a single node, we also run multiple nodes in memory, even within the very same test.

Reply:
It cannot fundamentally be safe in current form because it is racey to assume the state of the map will not change before you you access it later. You would have to hold the lock the whole time. Regardless, this approach doesn't make sense because there is no reason to share global state and it only introduces a burden. Entangling isolated systems doesn't help us here.

Reply:
Yeah fair enough, I'll refactor this and let you know when it's ready for another review.